src/adjoint_common.jl

```diff
@@ -419,8 +424,12 @@ function (f::ReverseLossCallback)(integrator)
     if F !== nothing
         F !== I && F !== (I, I) && ldiv!(F, Δλd)
     end
-
-    u[diffvar_idxs] .+= Δλd
+    
+    if u isa StaticArrays.SArray
+        integrator.u += Δλd
```

Comment:
comment that this drops diffvar_idxs because it assumes QuadratureAdjoint, or even better, put an `@assert` in here for that.